safe/common/parameters/group_select_parameter_widget.py

```diff
@@ -0,0 +1,204 @@
+        self.list_widget.setDefaultDropAction(Qt.MoveAction)
+        self.list_widget.setEnabled(False)
+
+        for i, key in enumerate(self._parameter.options):
+            value = self._parameter.options[key]
+            radio_button = QRadioButton(value.get('label'))
+            self.radio_button_layout.addWidget(radio_button, i, 0)
+            if value.get('type') == SINGLE_DYNAMIC:
+                double_spin_box = QDoubleSpinBox()
+                self.radio_button_layout.addWidget(double_spin_box, i, 1)
+                double_spin_box.setValue(value.get('value', 0))
+                double_spin_box.setMinimum(value.get('constraint', {}).get(
+                    'min', 0))
+                double_spin_box.setMaximum(value.get('constraint', {}).get(
+                    'max', 1))
+                double_spin_box.setSingleStep(value.get('constraint', {}).get(
+                    'step', 0.01))
+                self.spin_boxes[key] = double_spin_box
+
+                # Enable spin box depends on the selected option
+                if self._parameter.selected == key:
+                    double_spin_box.setEnabled(True)
+                else:
+                    double_spin_box.setEnabled(False)
+
+            elif value.get('type') == STATIC:
+                static_value = value.get('value', 0)
+                if static_value is not None:
+                    self.radio_button_layout.addWidget(
+                        QLabel(str(static_value)), i, 1)
+            elif value.get('type') == MULTIPLE_DYNAMIC:
+                selected_fields = value.get('value', [])
+                if self._parameter.selected == key:
+                    self.list_widget.setEnabled(True)
+                else:
+                    self.list_widget.setEnabled(False)
+
+            self.input_button_group.addButton(radio_button, i)
+            if self._parameter.selected == key:
+                radio_button.setChecked(True)
+
+        self.inner_input_layout.addLayout(self.radio_button_layout)
+        self.inner_input_layout.addWidget(self.list_widget)
+
+        # Put elements into layouts
+        self.input_layout.addWidget(self.label)
+        self.input_layout.addLayout(self.inner_input_layout)
+        self.input_layout.setStretch(0, 1)
+        self.input_layout.setStretch(1, 0)
+
+        self.main_layout.addLayout(self.input_layout)
+        self.main_layout.addSpacerItem(QSpacerItem(0, 0))
+
+        self.setLayout(self.main_layout)
+
+        self.setSizePolicy(QSizePolicy.Minimum, QSizePolicy.Fixed)
+
+        # Update list widget
+        self.update_list_widget()
+
+        # Connect signal
+        self.input_button_group.buttonClicked.connect(
+            self.radio_buttons_clicked)
+
+    def get_parameter(self):
+        """Obtain list parameter object from the current widget state.
+        :returns: A DefaultValueParameter from the current state of widget
```

Comment:
I'm confused. We always add a new line @ismailsunni ? 
I went to check our wiki : https://github.com/inasafe/inasafe/wiki/Coding-Standards#doc-strings-and-comments It's correct.
I just edited the wiki for what I know on landscape : https://github.com/inasafe/inasafe/wiki/Coding-Standards#landscape Maybe I'm becoming landscape addict too now :D